### ISSUE TITLE
how does loss_sprior work?

### ISSUE THREAD

**Issue by billyzs** [Digital-Humans-23/a3#8]
loss_sprior is always zero, which doesn't feel right. Looking at the code, it's calculatedfrom
```
loss_sprior = self.weight_sprior * torch.mean(betas**2)
```
but we set
```
betas = torch.zeros(1, 12).to(self.device)
```
so this doesn't feel right. But I'm not sure how to modify it.

Can you link the LISST paper? I don't think the slides contain enough information to solve this assignment. For example, the slides make no mention of what is global v.s. what is local loss.

**Comment by yz-cnsdqz** [comment]
two possible reasons:
- is `betas.requires_grad=True`? 
- is `self.weight_sprior=0`? 

There is no paper for LISST yet. Global configurations mean the root/pelvis location and orientation, with which you can place the body in the world coordinate. Local configurations mean the body shape and pose, with which you can pose the body in the body coordinate. 

There is no clear definition in the community as well.

**Comment by Milkiananas** [comment]
Notice that `weight_sprior` is set to zero by default

**Comment by billyzs** [comment]
@yz-cnsdqz 

My code passes the test but I find the concept of early v.s. late stage training to be confusing. I'm assuming the intention is that in early stage of training, we mask `loss_pprior` and `loss_sprior` to be either very small or zero; then after a chosen number of episodes we unmask those values so that they have more influence over the total loss. But `loss_pprior` and `loss_sprior` are already very small to begin with, even if I set their respective weights to 1,  and start to turn on these 2 losses after 10% of iterations, I get

```
[iter_inner= 0] PROJ=176.800, SPRIOR=0.000, PPRIOR=0.000, SMOOTH=0.000, TIME=0.51, FACTOR=0.000
[iter_inner= 2] PROJ=145.109, SPRIOR=0.000, PPRIOR=0.000, SMOOTH=0.000, TIME=0.41, FACTOR=0.000
...
[iter_inner=1998] PROJ=5.801, SPRIOR=0.001, PPRIOR=0.213, SMOOTH=0.000, TIME=0.43, FACTOR=1.000
[iter_inner=1999] PROJ=5.801, SPRIOR=0.001, PPRIOR=0.213, SMOOTH=0.000, TIME=0.39, FACTOR=1.000
```
and it still passes the test.

In other words, the reprojection loss completely outweights the other loss terms;  If this is the case, why bother with turning on/off those losses at all?

**Comment by yz-cnsdqz** [comment]
As mentioned, you can implement early/late stage in several ways. You can either mask the losses, or you keep the losses there, but disable the relevant parameters to update via back-propagation. 

There is no unique way to implement the same thing, and this is what I expect students to explore. Before you set the loss weights low, you never foresee how the priors will influence the results, right?
